middleware.go

```diff
@@ -166,8 +166,13 @@ func parseRequestBody(c *Client, r *Request) (err error) {
 
 CL:
 	// by default resty won't set content length, you can if you want to :)
-	if (c.setContentLength || r.setContentLength) && r.bodyBuf != nil {
-		r.Header.Set(hdrContentLengthKey, fmt.Sprintf("%d", r.bodyBuf.Len()))
+	if c.setContentLength || r.setContentLength {
+		if r.bodyBuf != nil {
+			r.Header.Set(hdrContentLengthKey, fmt.Sprintf("%d", r.bodyBuf.Len()))
+		} else {
+			r.Header.Set(hdrContentLengthKey, "0")
+
+		}
```

Comment:
@praem90 when we have true & false block, it better to go for true block first.
```go
if r.bodyBuf == nil {
	r.Header.Set(hdrContentLengthKey, "0")
} else {
	r.Header.Set(hdrContentLengthKey, fmt.Sprintf("%d", r.bodyBuf.Len()))
}
```

Reply:
Its good to know the little trick and fixed it.